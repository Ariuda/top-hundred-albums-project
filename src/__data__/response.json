{
    "feed": {
        "entry": [
            {
                "id": {
                    "attributes": {
                        "im:id": "999999"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999998"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999997"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999996"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999995"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999994"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999993"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999992"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999991"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999990"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999989"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999988"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999987"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999986"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999985"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999984"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999983"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999982"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999981"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999980"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999979"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999978"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999977"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999976"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999975"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999974"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999973"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            },
            {
                "id": {
                    "attributes": {
                        "im:id": "9999972"
                    }
                },
                "im:name": {
                    "label": "Album title"
                }, 
                "im:artist": { 
                    "label": "Album author", 
                    "attributes": {
                        "href": "link"
                    }
                },
                "im:image": [
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    },
                    {
                        "label": "image-string"
                    }
                ],
                "im:price": {
                    "label": "album-price"
                },
                "category": {
                    "attributes": {
                        "label": "category",
                        "scheme": "category-link"
                    }
                }
            }
        ] 
    }
}
