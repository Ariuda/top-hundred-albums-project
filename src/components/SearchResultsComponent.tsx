import React from 'react';
import { Link } from 'react-router-dom';
import { Album } from '../entities';

interface Props {
    results: Album[];
    onClick: () => void;
}

export const SearchResultsComponent: React.FC<Props> = ({ results, onClick }) => {

    const renderResults = () => {
        if (results.length > 0) {
            return results.map(result => {
                return <li key={result.id} onClick={onClick}><Link to={`/album/${result.id}`}>{result.title}, by {result.artist.name} - {result.category.name}</Link></li>
            });
        };
        return <li className="no-result">Your search did not match any results</li>;
    }

    return (
        <ul>
            {renderResults()}
        </ul>
    )
};

export default SearchResultsComponent;