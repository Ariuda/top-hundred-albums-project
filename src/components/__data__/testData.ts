export const testData = [
    {
        id: '123',
        link: 'link',
        title: 'title 1',
        artist: {
            name: 'name 1',
            link: 'link-one',
        },
        image: 'image-one',
        price: 'price-one',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 1,
        favorite: false  
    },
    {
        id: '124',
        link: 'link',
        title: 'title 2',
        artist: {
            name: 'name 2',
            link: 'link',
        },
        image: 'image',
        price: 'price-two',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 2,
        favorite: false   
    }
];