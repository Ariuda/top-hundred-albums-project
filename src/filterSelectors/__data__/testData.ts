export const testData = [
    {
        id: '123',
        link: 'link',
        title: 'title 1',
        artist: {
            name: 'name 1',
            link: 'link-one',
        },
        image: 'image-one',
        price: 'price-one',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 1,
        favorite: false  
    },
    {
        id: '124',
        link: 'link',
        title: 'title 2',
        artist: {
            name: 'name 2',
            link: 'link',
        },
        image: 'image',
        price: 'price-two',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 2,
        favorite: false   
    },
    {
        id: '125',
        link: 'link',
        title: 'title 3',
        artist: {
            name: 'name 3',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 3,
        favorite: false   
    },
    {
        id: '126',
        link: 'link',
        title: 'title 4',
        artist: {
            name: 'name 4',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 4,
        favorite: false   
    },
    {
        id: '126',
        link: 'link',
        title: 'title 5',
        artist: {
            name: 'name 5',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 5,
        favorite: false   
    },
    {
        id: '127',
        link: 'link',
        title: 'title 6',
        artist: {
            name: 'name 6',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 6,
        favorite: false   
    },
    {
        id: '128',
        link: 'link',
        title: 'title 7',
        artist: {
            name: 'name 7',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 7,
        favorite: false   
    },
    {
        id: '129',
        link: 'link',
        title: 'title 8',
        artist: {
            name: 'name 8',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 8,
        favorite: false   
    },
    {
        id: '130',
        link: 'link',
        title: 'title 9',
        artist: {
            name: 'name 9',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 9,
        favorite: false   
    },
    {
        id: '131',
        link: 'link',
        title: 'title 10',
        artist: {
            name: 'name 10',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 10,
        favorite: false   
    },
    {
        id: '132',
        link: 'link',
        title: 'title 11',
        artist: {
            name: 'name 11',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 11,
        favorite: false   
    },
    {
        id: '133',
        link: 'link',
        title: 'title 12',
        artist: {
            name: 'name 12',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 12,
        favorite: false   
    },
    {
        id: '134',
        link: 'link',
        title: 'title 13',
        artist: {
            name: 'name 13',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 13,
        favorite: false   
    },
    {
        id: '135',
        link: 'link',
        title: 'album 14',
        artist: {
            name: 'name 14',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 14,
        favorite: false   
    },
    {
        id: '136',
        link: 'link',
        title: 'different title 15',
        artist: {
            name: 'name 15',
            link: 'link',
        },
        image: 'image',
        price: 'price',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 15,
        favorite: false   
    }
];