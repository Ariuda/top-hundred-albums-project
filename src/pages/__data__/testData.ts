import { Album } from "../../entities";

export const testData: Album = {
        id: '123',
        link: 'link',
        title: 'title 1',
        artist: {
            name: 'name 1',
            link: 'link-one',
        },
        image: 'image-one',
        price: 'price-one',
        category: {
            name: 'category',
            link: 'category-link'
        },
        numberOfItems: '8',
        releaseDate: 'release-date',
        rights: 'rights-data',
        index: 1,
        favorite: false  
    };