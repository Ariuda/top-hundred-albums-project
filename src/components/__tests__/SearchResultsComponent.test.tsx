import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { SearchResultsComponent } from '../SearchResultsComponent';
import { testData } from '../__data__/testData';

const props = {
    onClick: () => console.log('test')
}

it('should render the search results if there are any results', () => {
    const wrapper = shallow(
        <SearchResultsComponent 
            results={testData}
            onClick={props.onClick}
        />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
});

it('should not render the search results if there are no results', () => {
    const wrapper = shallow(
        <SearchResultsComponent 
            results={[]}
            onClick={props.onClick}
        />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
});