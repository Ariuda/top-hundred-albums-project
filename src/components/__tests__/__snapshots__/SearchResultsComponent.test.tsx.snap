// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render the search results if there are no results 1`] = `
<ul>
  <li
    className="no-result"
  >
    Your search did not match any results
  </li>
</ul>
`;

exports[`should render the search results if there are any results 1`] = `
<ul>
  <li
    key="123"
    onClick={[Function]}
  >
    <Link
      to="/album/123"
    >
      title 1
      , by 
      name 1
       - 
      category
    </Link>
  </li>
  <li
    key="124"
    onClick={[Function]}
  >
    <Link
      to="/album/124"
    >
      title 2
      , by 
      name 2
       - 
      category
    </Link>
  </li>
</ul>
`;
